From dde57bab7a306e722f3e9827a293fb102122196c Mon Sep 17 00:00:00 2001
From: Eli Nidam <elini@marvell.com>
Date: Sun, 29 Dec 2013 15:20:16 +0200
Subject: [PATCH 1249/1825] a38x: support SGMII module

https://github.com/MISL-EBU-System-SW/misl-windriver.git linux-3.4.69-14t2-read
commit 9c60e177cdccb0f1d2935cc24ac8e56c3cd478dd

	1. Remove unused subroutine mvBoardIsQsgmiiModuleConnected
	2. Update the SGMII board subroutine depend on SGMII module

Change-Id: I83ba55136f49e66e2c1ecc36ae10a39868f7288d
Signed-off-by: Eli Nidam <elini@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/4918
Reviewed-by: Omri Itach <omrii@marvell.com>
Tested-by: Omri Itach <omrii@marvell.com>
Signed-off-by: Zhong Hongbo <hongbo.zhong@windriver.com>
---
 .../armada_38x_family/boardEnv/mvBoardEnvLib.c     |   46 ++++++--------------
 .../armada_38x_family/boardEnv/mvBoardEnvLib.h     |    8 ++-
 .../armada_38x_family/boardEnv/mvBoardEnvSpec.c    |    1 +
 3 files changed, 19 insertions(+), 36 deletions(-)

diff --git a/arch/arm/mach-armada38x/armada_38x_family/boardEnv/mvBoardEnvLib.c b/arch/arm/mach-armada38x/armada_38x_family/boardEnv/mvBoardEnvLib.c
index 273c301..46ee299 100644
--- a/arch/arm/mach-armada38x/armada_38x_family/boardEnv/mvBoardEnvLib.c
+++ b/arch/arm/mach-armada38x/armada_38x_family/boardEnv/mvBoardEnvLib.c
@@ -270,14 +270,10 @@ MV_STATUS mvBoardNameGet(char *pNameBuff, MV_U32 size)
 *******************************************************************************/
 MV_BOOL mvBoardIsPortInSgmii(MV_U32 ethPortNum)
 {
-	MV_U8 readValue;
-
-	if (mvBoardTwsiGet(BOARD_TWSI_MODULE_DETECT, 2, 0, &readValue) != MV_OK) {
-		DB(mvOsPrintf("%s: Error: Read from TWSI failed\n", __func__));
-		return MV_FALSE;
-	}
-	if (readValue == 0x0d)
+	if ((mvBoardIsModuleConnected(MV_CONFIG_SGMII)) && /* when SGMII connected the SGMII ports are 1 & 2 */
+	    ((ethPortNum == 1) || (ethPortNum == 2)))
 		return MV_TRUE;
+
 	return MV_FALSE;
 }
 
@@ -301,6 +297,10 @@ MV_BOOL mvBoardIsPortInSgmii(MV_U32 ethPortNum)
 *******************************************************************************/
 MV_BOOL mvBoardIsPortInGmii(MV_U32 ethPortNum)
 {
+	/* On DB board currently when SGMII connected the SGMII module activet port 1 and port 2
+	   and port 0 is disabled */
+	if ((mvBoardIsModuleConnected(MV_CONFIG_MII)) && (ethPortNum == 0))
+		return MV_TRUE;
 	return MV_FALSE;
 }
 /*******************************************************************************
@@ -323,6 +323,8 @@ MV_BOOL mvBoardIsPortInGmii(MV_U32 ethPortNum)
 *******************************************************************************/
 MV_BOOL mvBoardIsPortInRgmii(MV_U32 ethPortNum)
 {
+	if (mvBoardIsPortInGmii(ethPortNum) || mvBoardIsPortInSgmii(ethPortNum))
+		return MV_FALSE;
 	if (ethPortNum < 2)
 		return MV_TRUE;
 	return MV_FALSE;
@@ -1304,6 +1306,7 @@ MV_VOID mvBoardMppModuleTypePrint(MV_VOID)
 		"NOR 16bit",                                  \
 		"NAND 16bit",                                 \
 		"SDIO 4bit",                                 \
+		"SGMII",                                 \
 	};
 	mvOsOutput("Board configuration detected:\n");
 
@@ -1336,6 +1339,8 @@ MV_VOID mvBoardOtherModuleTypePrint(MV_VOID)
 *******************************************************************************/
 MV_BOOL mvBoardIsGbEPortConnected(MV_U32 ethPortNum)
 {
+	if ((mvBoardIsModuleConnected(MV_CONFIG_SGMII)) && (ethPortNum == 0))
+		return MV_FALSE;
 	if (ethPortNum < mvCtrlEthMaxPortGet())
 		return MV_TRUE;
 
@@ -2097,10 +2102,7 @@ MV_U32 mvBoardSwitchCpuPortGet(MV_U32 switchIdx)
 *******************************************************************************/
 MV_BOOL mvBoardIsEthConnected(MV_U32 ethNum)
 {
-	if ((ethNum == 0) || (ethNum == 1))
-		return MV_TRUE;
-
-	return MV_FALSE;
+	return mvBoardIsGbEPortConnected(ethNum);
 }
 
 /*******************************************************************************
@@ -2122,32 +2124,10 @@ MV_BOOL mvBoardIsEthConnected(MV_U32 ethNum)
 *******************************************************************************/
 MV_BOOL mvBoardIsEthActive(MV_U32 ethNum)
 {
-	/* for A375, all connected ports are Active and usabe */
 	return mvBoardIsEthConnected(ethNum);
 }
 
 /*******************************************************************************
-* mvBoardIsQsgmiiModuleConnected
-*
-* DESCRIPTION:
-*       This routine returns whether the QSGMII module is connected or not.
-*
-* INPUT:
-*       None.
-*
-* OUTPUT:
-*       None.
-*
-* RETURN:
-*       MV_TRUE if QSGMII module is connected, MV_FALSE otherwise.
-*
-*******************************************************************************/
-MV_BOOL mvBoardIsQsgmiiModuleConnected(MV_VOID)
-{
-	return MV_FALSE;
-}
-
-/*******************************************************************************
 * mvBoardGePhySwitchPortGet
 *
 * DESCRIPTION:
diff --git a/arch/arm/mach-armada38x/armada_38x_family/boardEnv/mvBoardEnvLib.h b/arch/arm/mach-armada38x/armada_38x_family/boardEnv/mvBoardEnvLib.h
index 8f2b0fa..47d64ea 100644
--- a/arch/arm/mach-armada38x/armada_38x_family/boardEnv/mvBoardEnvLib.h
+++ b/arch/arm/mach-armada38x/armada_38x_family/boardEnv/mvBoardEnvLib.h
@@ -184,9 +184,10 @@ typedef enum _mvConfigTypeID {
 	MV_CONFIG_NOR			= BIT4,	/* NOR board SLM 1361	*/
 	MV_CONFIG_NAND			= BIT5,	/* NAND board SLM 1361	*/
 	MV_CONFIG_SDIO			= BIT6,	/* SDIO board SLM 1361	*/
-	MV_CONFIG_NAND_ON_BOARD		= BIT7,	/* ON board nand detect */
-	MV_CONFIG_TYPE_MAX_MODULE	= 7,
-	MV_CONFIG_TYPE_MAX_OPTION	= 8
+	MV_CONFIG_SGMII			= BIT7,	/* SDIO board SLM 1364	*/
+	MV_CONFIG_NAND_ON_BOARD		= BIT8,	/* ON board nand detect */
+	MV_CONFIG_TYPE_MAX_MODULE	= 8,
+	MV_CONFIG_TYPE_MAX_OPTION	= 9
 } MV_CONFIG_TYPE_ID;
 
 typedef struct _devCsInfo {
@@ -392,6 +393,7 @@ typedef struct _boardInfo {
 { MV_CONFIG_NOR,		0x4,	0,	 0xF,	{ 0, 1, 0, 0} }, \
 { MV_CONFIG_NAND,		0x4,	0,	 0x1,	{ 0, 1, 0, 0} }, \
 { MV_CONFIG_SDIO,		0x4,	0,	 0x2,	{ 0, 1, 0, 0} }, \
+{ MV_CONFIG_SGMII,		0x2,	0,	 0xF,	{ 0, 1, 0, 0} }, \
 };
 struct _mvBoardMppModule {
 	MV_U32 group;
diff --git a/arch/arm/mach-armada38x/armada_38x_family/boardEnv/mvBoardEnvSpec.c b/arch/arm/mach-armada38x/armada_38x_family/boardEnv/mvBoardEnvSpec.c
index a43ce7d..fa8d22a 100644
--- a/arch/arm/mach-armada38x/armada_38x_family/boardEnv/mvBoardEnvSpec.c
+++ b/arch/arm/mach-armada38x/armada_38x_family/boardEnv/mvBoardEnvSpec.c
@@ -109,6 +109,7 @@ MV_BOARD_MAC_INFO db88f68xxInfoBoardMacInfo[] = {
 	/* {{MV_BOARD_MAC_SPEED boardMacSpeed, MV_8 boardEthSmiAddr}} */
 	{ BOARD_MAC_SPEED_AUTO, 0},
 	{ BOARD_MAC_SPEED_AUTO, 0x1},
+	{ BOARD_MAC_SPEED_AUTO, 0x0},
 };
 
 MV_DEV_CS_INFO db88f68xxInfoBoardDeCsInfo[] = {
-- 
1.7.5.4

